res/controllers/Numark-Mixtrack-Pro-FX-scripts.js

```diff
@@ -0,0 +1,772 @@
+        if (component.group === undefined) {
+            component.group = this.currentDeck;
+        }
+    });
+};
+
+MixtrackProFX.Deck.prototype = new components.Deck();
+
+MixtrackProFX.Browse = function() {
+    this.knob = new components.Encoder({
+        group: "[Library]",
+        inKey: "Move",
+        input: function(channel, control, value) {
+            if (value === 0x01) {
+                engine.setParameter(this.group, this.inKey + "Down", 1);
+            } else if (value === 0x7F) {
+                engine.setParameter(this.group, this.inKey + "Up", 1);
+            }
+        }
+    });
+
+    this.knobShift = new components.Encoder({
+        input: function(channel, control, value) {
+            if (value === 0x01) {
+                engine.setParameter("[Channel1]", "waveform_zoom_down", 1);
+
+                // need to zoom both channels if waveform sync is disabled in Mixxx settings.
+                // and when it's enabled then no need to zoom 2nd channel, as it will cause
+                // the zoom to jump 2 levels at once
+                if (!MixtrackProFX.waveformsSynced) {
+                    engine.setParameter("[Channel2]", "waveform_zoom_down", 1);
+                }
+            } else if (value === 0x7F) {
+                engine.setParameter("[Channel1]", "waveform_zoom_up", 1);
+
+                // see above comment
+                if (!MixtrackProFX.waveformsSynced) {
+                    engine.setParameter("[Channel2]", "waveform_zoom_up", 1);
+                }
+            }
+        }
+    });
+
+    this.knobButton = new components.Button({
+        group: "[Library]",
+        inKey: "MoveFocusForward"
+    });
+
+    this.buttonShift = new components.Button({
+        group: "[Library]",
+        inKey: "GoToItem"
+    });
+
+    this.setBeatgrid = new components.Button({
+        group: "[Channel1]",
+        key: "beats_translate_curpos",
+        midi: [0x88, 0x01],
+        off: 0x01,
+        shift: function() {
+            this.group = "[Channel2]";
+        },
+        unshift: function() {
+            this.group = "[Channel1]";
+        }
+    });
+};
+
+MixtrackProFX.Browse.prototype = new components.ComponentContainer();
+
+MixtrackProFX.HeadGain = function() {
+    components.Pot.call(this);
+};
+
+MixtrackProFX.HeadGain.prototype = new components.Pot({
+    group: "[Master]",
+    inKey: "headGain"
+});
+
+MixtrackProFX.vuCallback = function(value, group) {
+    var level = value * 90;
+
+    if (engine.getValue("[Channel1]", "pfl") || engine.getValue("[Channel2]", "pfl")) {
+        if (group === "[Channel1]") {
+            midi.sendShortMsg(0xB0, 0x1F, level);
+        } else if (group === "[Channel2]") {
+            midi.sendShortMsg(0xB1, 0x1F, level);
+        }
+    } else if (group === "[Channel1]") {
+        midi.sendShortMsg(0xB0, 0x1F, level);
+    } else if (group === "[Channel2]") {
+        midi.sendShortMsg(0xB1, 0x1F, level);
+    }
```

Comment:
this seems unnecessarily complicated to me. How about this:
```suggestion
    var level = value * 90;
    var deckOffset = script.deckFromGroup(group) - 1;
    midi.sendShortMsg(0xB0 + deckOffset, 0x1F, level);
```